src/librustc_mir/shim.rs

```diff
@@ -259,6 +272,164 @@ impl<'a, 'tcx> DropElaborator<'a, 'tcx> for DropShimElaborator<'a, 'tcx> {
     }
 }
 
+/// Build a `Clone::clone` shim for `recvr_ty`. Here, `def_id` is `Clone::clone`.
+fn build_clone_shim<'a, 'tcx>(tcx: ty::TyCtxt<'a, 'tcx, 'tcx>,
+                              def_id: DefId,
+                              rcvr_ty: ty::Ty<'tcx>)
+                              -> Mir<'tcx>
+{
+    let sig = tcx.fn_sig(def_id);
```

Comment:
@arielb1 how would we handle bounds checking? `#[derive(Clone)]` currently uses local variables of type `AssertParamIsClone<T: Clone + ?Sized>` for each field type `T`  in order to deal with recursive types. This would not be applicable here.